flink-table/flink-table-planner-blink/src/test/scala/org/apache/flink/table/planner/runtime/stream/sql/WindowAggregateITCase.scala

```diff
@@ -347,6 +347,83 @@ class WindowAggregateITCase(mode: StateBackendMode)
     assertEquals(expected.sorted, sink.getAppendResults.sorted)
   }
 
+  @Test
+  def testEventTimeTumblingWindowWithOffset(): Unit = {
+    val stream = failingDataSource(data)
+      .assignTimestampsAndWatermarks(
+        new TimestampAndWatermarkWithOffset
+          [(Long, Int, Double, Float, BigDecimal, String, String)](10L))
+    val table = stream.toTable(tEnv,
+      'rowtime.rowtime, 'int, 'double, 'float, 'bigdec, 'string, 'name)
+    tEnv.registerTable("T1", table)
+
+    val sql =
+      """
+        |SELECT
+        |  `string`,
+        |  TUMBLE_START(rowtime, INTERVAL '0.005' SECOND, TIME '00:00:00.003'),
```

Comment:
The `offset` parameter type is `TIME` in Calcite by default. Then we cannot support offsets larger than one day? E.g., we want to have a 1 week tumble window from Monday to Sunday, then we need `TUMBLE(ts, INTERVAL 7 DAY, INTERVAL 3 DAY)`, current implementation cannot do this.

Reply:
Yes, you are right. `TIME` type cannot support offsets larger than one day. I think we should determine what type is reasonable to use. If `INTERVAL`  is reasonable, we can use `DATETIME_INTERVAL_INTERVA` and `DATETIME_INTERVAL_INTERVAL_INTERVAL`

Reply:
Personally, I prefer `INTERVAL` than `TIME`, `INTERVAL` is more powerful. WDYT? @wuchong ?

Reply:
Hi @libenchao , I'm not sure whether we can support `INTERVAL` here. Because AFAIK, the type has been defined by Calcite.

Reply:
@wuchong We can modify the definition of `TUMBLE`/`HOP`/`SESSION` in `FlinkSqlOperatorTable`, then we have the flexibility to define the operand types. (Actually I did this in 1.9 internally already)

Reply:
I use `INTERVAL` type in this PR https://github.com/apache/flink/compare/master...wangxlong:FLINK-17767_OFFSET_USE_INTERVAL?expand=1.

Reply:
Personally, I don't see a strong need for `INTERVAL`. I think the offset is used for time zone offset, so it shoudn't larger than day. 

cc @danny0405 , do you think the reason why Calcite use TIME type for the offset argument?